framework/connection.py

```diff
@@ -0,0 +1,78 @@
+"""
+Module to connect to remote host
+"""
+
+import logging
+
+from common_ci_utils.connection import Connection
+from framework import config
+from paramiko.auth_handler import AuthenticationException, SSHException
+
+log = logging.getLogger(__name__)
+
+
+class SSHConnection:
+    """
+    A class that connects to remote host
+    """
+
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        if not cls._instance:
+            cls._instance = super(SSHConnection, cls).__new__(cls)
+            cls._instance._conn = None
```

Comment:
```suggestion
```

Reply:
initializing the _conn here , otherwise it will fail at https://github.com/red-hat-storage/noobaa-sa-ci/pull/2/files#diff-9e97d86ebad152bc0a7b249d4b1f794a0b329fccc11aba53cf7001903241031cR37